sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,38 +522,96 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      // Here, this rule only resolves the missing sort references if the child is not Aggregate
+      //   Another rule ResolveAggregateFunctions will resolve that case.
```

Comment:
@cloud-fan I kept the function implementation in the `ResolveAggregateFunctions`, but I called the function in `ResolveSortReferences`. Since the rule `ResolveAggregateFunctions` covers two cases (`filter` and `sort`), I am afraid the code readers might feel confused if we split them into two rules. This function call is public. I am not sure if this way is appropriate?